pkg/server/adminHandler.go

```diff
@@ -399,24 +426,43 @@ func (h *adminHandler) deviceConfigSet(w http.ResponseWriter, r *http.Request) {
 }
 
 func (h *adminHandler) deviceLogsGet(w http.ResponseWriter, r *http.Request) {
-	h.deviceDataGet(w, r, h.logChannel, h.manager.GetLogsReader)
+	h.deviceDataGet(w, r, h.logChannel, h.manager.GetLogsReader, nil)
 }
 
 func (h *adminHandler) deviceInfoGet(w http.ResponseWriter, r *http.Request) {
-	h.deviceDataGet(w, r, h.infoChannel, h.manager.GetInfoReader)
+	h.deviceDataGet(w, r, h.infoChannel, h.manager.GetInfoReader, func(in []byte) ([]byte, error) {
+		var err error
+		msg := &info.ZInfoMsg{}
+		if err = proto.Unmarshal(in, msg); err != nil {
+			return nil, fmt.Errorf("error parsing info message: %v", err)
+		}
+		var entryBytes []byte
```

Comment:
We unmarshal from the proto bytes to the struct, then from the struct to json? Couldn't we just do it straight?